src/hotspot/share/opto/loopnode.cpp

```diff
@@ -855,6 +855,28 @@ bool PhaseIdealLoop::create_loop_nest(IdealLoopTree* loop, Node_List &old_new) {
     // not a loop after all
     return false;
   }
+
+  if (range_checks.size() > 0) {
+    // This transformation requires peeling one iteration. Also, if it has range checks and they are eliminated by
+    // Loop Predication, then 2 Hoisted Check Predicates are added for one range check. Finally, transforming a long range check requires
```

Comment:
Just noticed that the line wraps are off now - maybe you want to re-wrap these lines. But that's just a minor detail.